neptune/new/sync/status.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) 2022, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+__all__ = ["StatusRunner"]
+
+import sys
+import textwrap
+from pathlib import Path
+from typing import List, Sequence, Tuple
+
+import click
+
+from neptune.new.constants import (
+    ASYNC_DIRECTORY,
+    OFFLINE_NAME_PREFIX,
+)
+from neptune.new.envs import PROJECT_ENV_NAME
+from neptune.new.internal.backends.api_model import ApiExperiment
+from neptune.new.sync.abstract_backend_runner import AbstractBackendRunner
+from neptune.new.sync.utils import (
+    get_qualified_name,
+    is_run_synced,
+    iterate_containers,
+    get_metadata_container,
+    get_offline_dirs,
+)
+
+offline_run_explainer = """
+Runs which execute offline are not created on the server and they are not assigned to projects;
+instead, they are identified by UUIDs like the ones above.
+When synchronizing offline runs, please specify the workspace and project using the "--project"
+flag. Alternatively, you can set the environment variable
+{} to the target workspace/project. See the examples below.
+""".format(
+    PROJECT_ENV_NAME
+)
+
+
+class StatusRunner(AbstractBackendRunner):
+    def partition_containers(
+        self,
+        base_path: Path,
+    ) -> Tuple[List[ApiExperiment], List[ApiExperiment], int]:
+        synced_runs = []
+        unsynced_runs = []
+        async_path = base_path / ASYNC_DIRECTORY
+        for container_type, container_id, path in iterate_containers(async_path):
+            metadata_container = get_metadata_container(
+                container_id=container_id,
+                container_type=container_type,
+                backend=self._backend,
+            )
+
+            if is_run_synced(path):
```

Comment:
I have to say I really don't care for the way you're renamed some functions to e.g. `partition_containers`, yet kept others e.g. `is_run_synced`. But I guess it's going to be cleaned up in a future task.

Reply:
Yes, it will.